### DISCUSSION TITLE
Glue JDBC connection needs password in the console before it becomes valid

### DISCUSSION THREAD

**Discussion by krzysztof-slowinski** [aws/aws-cdk#13615] [Q&A]
I am creating a JDBC connection in Glue using secrets for username and password. I can see in the console that username is read correctly from the secret, so that's not a concern. Once I edit the details and enter the password in the console, it becomes valid. Is there something wrong with my approach?

```
glue.CfnConnection(
            self,
            id="JDBCConnection",
            catalog_id=self.account,
            connection_input=glue.CfnConnection.ConnectionInputProperty(
                name="jdbc_connection",
                connection_type="JDBC",
                physical_connection_requirements=glue.CfnConnection.PhysicalConnectionRequirementsProperty(
                    subnet_id=cdk.Fn.import_value("PrivateSubnet1"),
                    security_group_id_list=[jdbc_connection_security_group.attr_group_id],
                ),
                connection_properties={
                    "JDBC_CONNECTION_URL": "jdbc:<JDBC_URL>",
                    "USERNAME": "{{resolve:secretsmanager:jdbc_username}}",
                    "PASSWORD": "{{resolve:secretsmanager:jdbc_password}}",
                },
            ),
        )
```

**Comment by tim-finnigan** [comment]
Checking in on this discussion as it was opened a while ago. It looks like you also asked about this in Stack Overflow and received some replies: https://stackoverflow.com/questions/66657768/aws-glue-jdbc-connection-created-with-cdk-needs-password-in-the-console-before-i

[This answer](https://stackoverflow.com/a/69474820) on that thread was marked as the accepted solution, so I assume that resolved the issue. That user said the issue was due to missing the SSL and the availability zone, specifically the `ConnectionProperties` key `JDBC_ENFORNCE_SSL` and `PhysicalConnectionRequirements` key `AvailabilityZone`. 

For future reference I'll also share this knowledge center article on troubleshooting Glue connection issues as it might help someone who stumbles on this: https://aws.amazon.com/premiumsupport/knowledge-center/glue-test-connection-failed/
